pkg/test/server.go

```diff
@@ -0,0 +1,156 @@
+	return fmt.Sprintf("%v:%v", s.Addr, s.UDPPort)
+}
+
+func (s *Server) parseTCPAddr() string {
+	return fmt.Sprintf("%v:%v", s.Addr, s.TCPPort)
+}
+
+func (s *Server) Run() {
+	g := new(errgroup.Group)
+	udpSrv := &dns.Server{Addr: s.parseUDPAddr(), Net: "udp", Handler: &udpHandler{}}
+	defer func() {
+		err := udpSrv.Shutdown()
+		if err != nil {
+			logrus.Errorf("shutting down UDP server: %v", err)
+		}
+	}()
+	g.Go(func() error {
+		return udpSrv.ListenAndServe()
+	})
+	tcpSrv := &dns.Server{Addr: s.parseTCPAddr(), Net: "tcp", Handler: &tcpHandler{}}
+	defer func() {
+		err := tcpSrv.Shutdown()
+		if err != nil {
+			logrus.Errorf("shutting down TCP server: %v", err)
+		}
+	}()
+	g.Go(func() error {
+		return tcpSrv.ListenAndServe()
+	})
+
+	if err := g.Wait(); err != nil {
+		logrus.Fatalf("Failed starting servers: %v", err)
+	}
+}
+
+type tcpHandler struct{}
+
+func (h *tcpHandler) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
+	msg := new(dns.Msg)
+	msg.SetReply(r)
+	msg.Compress = false
+	// we might handle other OpCodes later
+	if r.Opcode != dns.OpcodeQuery {
+		return
+	}
+	parseReply(msg)
+	err := w.WriteMsg(msg)
+	if err != nil {
+		logrus.Errorf("ServeDNS trying to write message in tcpHandler: %v", err)
+	}
+}
+
+type udpHandler struct{}
+
+func (h *udpHandler) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
+	msg := new(dns.Msg)
+	msg.SetReply(r)
+	msg.Compress = false
+	// we might handle other OpCodes later
+	if r.Opcode != dns.OpcodeQuery {
+		return
+	}
+	parseReply(msg)
+	msg.Truncate(truncateSize)
+	err := w.WriteMsg(msg)
+	if err != nil {
+		logrus.Errorf("ServeDNS trying to write message in udpHandler: %v", err)
+	}
+}
+
+func parseReply(msg *dns.Msg) {
+	logrus.Infof("ServeDNS handling request: %v", msg.Question)
+	for _, q := range msg.Question {
+		switch q.Qtype { //nolint:gocritic
+		case dns.TypeA:
+			msg.Authoritative = true
+			domain := q.Name
+			addresses, ok := domainToIP[domain]
+			if ok {
+				for _, addr := range addresses {
+					msg.Answer = append(msg.Answer, &dns.A{
+						Hdr: dns.RR_Header{
+							Name:   domain,
+							Rrtype: dns.TypeA,
+							Class:  dns.ClassINET,
+							Ttl:    60,
+						},
+						A: net.ParseIP(addr),
+					})
+				}
+			}
+		}
+	}
+}
+
+func loadRecords(path string) map[string][]string {
+	f, err := os.Open(path)
+	if err != nil {
+		log.Panicf("opening file: %v err: %v", path, err)
+	}
+	defer func() {
+		err := f.Close()
+		if err != nil {
+			logrus.Errorf("loadRecords closing file: %v", err)
+		}
+	}()
+
+	csvReader := csv.NewReader(f)
+	data, err := csvReader.ReadAll()
+	if err != nil {
+		log.Panicf("reading csv file: %v err: %v", path, err)
+	}
+
+	var domain string
+	records := make(map[string][]string)
+	for _, line := range data {
+		var ips = []string{}
+		for j, field := range line {
+			switch j {
+			case 0:
+				domain = fmt.Sprintf("%s.", field)
+			case 1:
+				ips = append(ips, field)
+			case 2:
+				ips = append(ips, field)
+			}
```

Comment:
```suggestion
			if j == 0 {
				domain = fmt.Sprintf("%s.", field)
			} else {
				ips = append(ips, field)
			}
```
This doesn't make sense as a switch (and using it also limits it to two addresses for no reason).

Reply:
True, I had another column initially. I will cleanup.